component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/choreo/QueueWorker.java

```diff
@@ -49,8 +54,15 @@ public void run() {
             }
             ThreadPoolExecutor threadPoolExecutor = ((ThreadPoolExecutor) executorService);
             do {
-                String event = eventQueue.poll();
-                if (event != null) {
+                MetricEventBuilder eventBuilder = eventQueue.poll();
+                if (eventBuilder != null) {
+                    String event = null;
+                    try {
+                        Map<String, Object> eventMap = eventBuilder.build();
```

Comment:
Suggestion:
Since queue worker only for event hub, here can we try to publish string return by the builder. Because looks like the Queue worker try to format the payload. What if we give that control to the builder? So we can publish what builder return.

Reply:
like pass the builder to EventhubPublisher where it gets build?

Reply:
something like this
```
String event = eventBuilder.build();
client.sendEvent(event);
```

So the interface like this?
```
public interface MetricEventBuilder {

    public String build() throws MetricReportingException;
}
```

Reply:
ahh I get it u want event builder to handle the json transformation too. There are few concerns though
- Restricting MetricEventBuilder.build() to return String is not great for customizations
- Current architecture is to have the builder provide you the validated attributes and its the responsibility of the Metric.incrementCount() to compose the actual message. In this way unless someone wants to add a new field they can keep on using the defaultBuilder and construct the payload they want in Metric class. In the new suggested way you will have to always implement a new builder for your reporter plus we will have to generalize the return type of the builder